content/blog/page-view-counter/index.md

```diff
@@ -0,0 +1,273 @@
+
+To start, we open Visual Studio and create a new project. In the Create a new project dialog box, search for functions, choose the Azure Functions template, and select Next. I'm creating the project with the name *PageViewCounter*. 
+
+Here is where we need to define some concepts related to functions:
+
+- There's a drop-down that contains the technologies that will host our functions. As mentioned previously, we will use *Azure Functions v2 (.NET Core)*.
+- Functions also provide automatic ways of triggering our code, some of which communicate with other Azure Services (Queues triggers, IoT Triggers, Blob Triggers, etc.). We'll use the [**HTTP trigger**](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook?tabs=csharp) as we want our function to trigger whenever we enter a page.
+- Authorization level: Anonymous. We don't need fancy ways of doing authentication for this so we'll leave it at that for simplicity.
+ 
+ After creating the project, we are received with boilerplate code that does some simple processing of a request; where if the HTTP request contains a name parameter, the response will be a greeting, and otherwise the request will fail with an error message:
+
+
+```csharp
+[FunctionName("Function1")]
+    public static async Task<IActionResult> Run(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+        ILogger log)
+    {
+        log.LogInformation("C# HTTP trigger function processed a request.");
+        string name = req.Query["name"];
+        string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+        dynamic data = JsonConvert.DeserializeObject(requestBody);
+        name = name ?? data?.name;
+        return name != null
+            ? (ActionResult)new OkObjectResult($"Hello, {name}")
+            : new BadRequestObjectResult("Please pass a name on the query string or in the request body");
+    } 
+```
+
+From here we can also see that some of the settings that we chose when creating the project defined the attribute of the **Run()** method (which is the main code we want to run).
+
+There are some other files in the project that we'll look into later.
+
+If we build and run the project, a console window opens up and gives us some information about our locally-hosted service:
+
+```bash
+Application started. Press Ctrl+C to shut down.
+
+Http Functions:
+
+        Function1: [GET,POST] http://localhost:7071/api/Function1
+```
+
+If we enter that URL into a browser without a name parameter, it will ask us to pass a parameter name. If we do so (for instance, by entering *http://localhost:7071/api/Function1?name=nacho*), it will say hello pertinently.
```

Comment:
maybe show how to use `curl` to request this? (just a suggestion)

Reply:
I feel like it does not get simpler than testing it through the browser, but I'll mention curl or Postman.